bin/alluxio-stop.sh

```diff
@@ -78,11 +75,48 @@ stop_proxies() {
 }
 
 stop_worker() {
+  if [[ ! -z "${cache}" ]] ; then
+    echo "Cache directory: ${cache}"
+    mkdir -p "${cache}"
```

Comment:
@ggezer What do you think about overwriting the contents at `${cache}` if the directory exists already? Stale files in the cache would be pretty bad to accumulate via `cp`.


I would prefer to prompt the user for confirmation before overwriting directories. If we add the user prompt then we'd also need to add a `-y` flag to auto-approve so that this can still be automated too.

Reply:
@madanadit WDYT?

Reply:
on start, we should clear the contents of the directory (since we always make a complete snapshot) once copied to memory. on stop, the directory should be empty to begin with (under normal operation), and we should be okay to overwrite the directory (without a -y flag)

Both sequences begin with empty directory:

Sequence 1: start (no action, dir empty dir) > stop (write snapshot 1, dir non-empty) > start (clear snapshot, dir empty) > stop (write snapshot 2, dir non-empty)

Sequence 2: start (no action, dir empty dir) > start (no action, dir empty dir) > stop (write snapshot 1, dir non-empty) > stop (overwrite snapshot 1, dir non-empty)> start (clear snapshot, dir empty)

Reply:
Yeah I agree that we don't need a prompt unless otherwise requested.

@madanadit what should we do during start if the cache-dir is empty but worker dir is not? Should we just end up clearing the existing content, taking the cache-dir as authority?

Reply:
start.sh: (src = cache-dir, dst = worker-dir). for this direction an overwrite-if-src-not-empty makes sense. if src is empty, do not overwrite since we do not want to clear the Alluxio cache the first time a user starts using the -c option.

stop.sh: (src = worker-dir, dst = cache-dir).  for this direction an overwrite-always makes sense. if src is empty, for example if someone calls alluxio fs free on /) then we do want to clear the cache-dir as well.